client/java/src/main/java/com/linecorp/centraldogma/client/CentralDogmaRequestPreparation.java

```diff
@@ -0,0 +1,295 @@
+     *   <li>{@code "/**"} - find all files recursively</li>
+     *   <li>{@code "*.json"} - find all JSON files recursively</li>
+     *   <li>{@code "/foo/*.json"} - find all JSON files under the directory {@code /foo}</li>
+     *   <li><code>"/&#42;/foo.txt"</code> - find all files named {@code foo.txt} at the second depth level</li>
+     *   <li>{@code "*.json,/bar/*.txt"} - use comma to specify more than one pattern. A file will be matched
+     *                                     if <em>any</em> pattern matches.</li>
+     * </ul>
+     */
+    public CentralDogmaDiffsRequest diffs(String pathPattern) {
+        requireNonNull(pathPattern, "pathPattern");
+        return new CentralDogmaDiffsRequest(this, pathPattern);
+    }
+
+    /**
+     * Returns a new {@link CentralDogmaPreviewDiffsRequest} that is used to send
+     * {@link CentralDogma#getPreviewDiffs(String, String, Revision, Iterable)} request to the
+     * Central Dogma repository.
+     */
+    public CentralDogmaPreviewDiffsRequest previewDiffs(Change<?>... changes) {
+        requireNonNull(changes, "changes");
+        return new CentralDogmaPreviewDiffsRequest(this, ImmutableList.copyOf(changes));
+    }
+
+    /**
+     * Returns a new {@link CentralDogmaPreviewDiffsRequest} that is used to send
+     * {@link CentralDogma#getPreviewDiffs(String, String, Revision, Iterable)} request to the
+     * Central Dogma repository.
+     */
+    public CentralDogmaPreviewDiffsRequest previewDiffs(Iterable<? extends Change<?>> changes) {
+        requireNonNull(changes, "changes");
+        return new CentralDogmaPreviewDiffsRequest(this, changes);
+    }
+
+    /**
+     * Returns a new {@link CentralDogmaCommitRequest} that is used to send
+     * {@link CentralDogma#push(String, String, Revision, String, String, Markup, Iterable)} request to the
+     * Central Dogma repository.
+     */
+    public CentralDogmaCommitRequest commit(String summary, Change<?>... changes) {
+        requireNonNull(changes, "changes");
+        return new CentralDogmaCommitRequest(this, summary, ImmutableList.copyOf(changes));
+    }
+
+    /**
+     * Returns a new {@link CentralDogmaCommitRequest} that is used to send
+     * {@link CentralDogma#push(String, String, Revision, String, String, Markup, Iterable)} request to the
+     * Central Dogma repository.
+     */
+    public CentralDogmaCommitRequest commit(String summary, Iterable<? extends Change<?>> changes) {
+        requireNonNull(changes, "changes");
+        return new CentralDogmaCommitRequest(this, summary, changes);
+    }
+
+    /**
+     * Returns a new {@link CentralDogmaWatchingFileRequest} that is used to send
+     * {@link CentralDogma#watchFile(String, String, Revision, Query, long)} request to
+     * the Central Dogma repository or create a new
+     * {@link CentralDogma#fileWatcher(String, String, Query, Function, Executor)}.
+     */
+    public CentralDogmaWatchingFileRequest<?> watchingFile(String path) {
```

Comment:
I prefer `watch` for the name of these APIs. We might express the watching API with `watch()`. For  example:
```java
repo.watch(path)
    .timeout(timeout)
    .once(Revision); // This can replace the orignal `watch()`

var watcher = 
repo.watch(path)
    .timeout(timeout)
    .start(); // This can replace the orignal `newWatcher()`
```

Reply:
I have to distinguish `watch(path)` (file watch) and `watch(pathPattern)` (repository watch).
Do you have any suggestions for this?

Reply:
How about introducing `PathPattern` API? https://github.com/line/centraldogma/pull/651#discussion_r762017789

Reply:
So the benefit of introducing the new `PathPattern` interface is
- using the same method name e.g. `watch(String)`, `watch(PathPattern)`
- Specifying multiple pathPattern (even though I'm not sure how we concat the string using a special character which shouldn't be used in the pathPattern)
Is there anything I missed except those?

Reply:
Actually, I’m not 100% sure that `PathPattern` is a good solution. But I guess we might provide more constant and fluent API with `PathPattern`.
If it is not your favorite, don't need to take the suggestion. 😄

Reply:
I like most of your suggestions but I just wanted to clarify if there's something that I missed so I left questions. 😆

Reply:
I still leaning toward using different method names because the returned instances are different depending on it is a file or files.
If `diffs` is vague how about appending files?
```java
repo.file(path);
repo.files(pathPattern);
repo.diff(path);
repo.diffFiles(pathPattern);
repo.diffFiles(Change...);
repo.watch(path);
repo.watchFiles(pathPattern);
```

Reply:
Sounds good to me except for `repo.diffFiles(Change...)`. Because technically, `Change` is a are not written to a file yet.

Reply:
On second thought, I think it's good to use `PathPattern` interface because we can put the scattered Javadoc description into one place and we can validate the path pattern when users put a comma into a path pattern by mistake.
Thanks!

Reply:
I guess `under` factory is also useful. 😆
```java
interface Pattern {
     Pattern under(String path) {
        return path + "/**";
     }
}

Pattern.under("/foo/bar");
```

Reply:
Let me create an issue for adding useful APIs to `PathPattern`. 😆

Reply:
Agreed.